### DISCUSSION TITLE
v1.6.4

### DISCUSSION THREAD

**Discussion by deniscerri** [deniscerri/ytdlnis#229] [General]
## Notice

Its suggested you backup your settings and history in case the app crashes on update, because this is a huge update.

## What's New

- Added Preferred Audio Format ID
- fixed korean bad strings
- fixed please wait card not showing up
- fixed audio format using mp4 containers
- added portugese language
- added punjabi language
- added greek language
- fixed container not updating in multi select card
- fixed swipe to delete bug in the queued section
- added download now option when u highlight only scheduled items in the context menu
- tapping on the errored notification will send you to errored tab, if you have logs disabled. If logs on it will send u to the log
- added preferred format ordering. By ID, FILESIZE, CONTAINER. Formats grouped by container will also be sorted by filesize
- fixed app not showing formats when u try to modify a current download item and its a different type (audio formats not showing if its a video type and vice versa)
- fixed app showing generic formats in cases where the format length was the same as that of generic formats (silly mistake)
- made the app store static strings for 'best' and 'worst' so that there isnt any confusion when u try to change the language and the stored downloads will have the other language's string
- added collected filesize on top of the download queue
- potentially fixed app not crashing when going to queued screen? idk
- fixed app not moving files when its a fresh install and u havent tried to change the download path.
- added download retries options [--retries and --fragment-retries]
- made the download logs not freeze the app
- download logs will work on quick downloaded items that were later updated
- added app arch and build type in the about section
- added "force-keyframes-at-cuts" preference
- Added download type in logs
- fix video not showing in cut screen. This depends on the streaming url
- fixed xhamster not showing results
- long press download button to save it for later and not schedule it. Also works when u try to queue multiple items
- change app icon background based on theme
- fixed bug when trying to redownload a history item and wanting to use a different type
- made the scheduled item auto-update after its queued so the app doesnt have to update when it begins downloading
- some fixes with output templates
- fixed container and vcodec being saved with translated strings and not constants
- added download thumbnail functionality. Click the result card in the middle and use the video player. Also observe running and queued downloads for that item
- added feature to disable thumbnails on certain screens. U can choose of course
- added feature to convert subs to different formats
- added youtube music search provider
- made app name have a color depending on the theme
- fixed format card not showing a translated string on best and worst strings
- added ability to hide the terminal from the share menu
- fixed app killing active downloads when removing a queued item
- fixed filename template not showing uploader on odysee
- added orange theme (its like yellow but slightly darker)
- fixed cookies not getting saved on older droids
- fixed app making new folder instead of merging when moving files
- added ability to put multiple preferred format ids separated by commas for both audio and video
- added support for piped links and treat them like youtube links
- fixed bug where if an item has no formats and you update audio formats and then go to video tab it will not show the video formats too
- made from and to textboxes wider in the cut video screen depending on the timestamp length
- added search engines in the search view
- removed filename template override if the user leaves it empty. Now it follows the yt-dlp default
- fixed app crashing sometimes when its tablet 
- added drag handle to scrolling content
- added pause functionality [This is somewhat wonky because the YT-DLP python process doesnt finish quick enough. So dont spam pause resume]
- made video player faster to load for youtube videos
- Added option to turn off usage of cookies
- some fixes with the cut screen
- Added ability to save command templates as extra commands by default on every downloadcard/AddExtraCommandsDialog
- fixed app not saving the proper youtube link and having to refetch data unnecessarily
- fixed app scrolling in the tablayout in the download queue screen while you are dragging the vertical scrollbar
- fixed app not destroying active downloads when you terminate the app
- added rewind button in the cut section
- fixed play button appearing in cut section when the video is playing
- removed "Downloading" word in the preferences screen. Its Downloads now
- Added embed metadata preference. Turning it off will remove any embedding metadata and parsing metadata commands
- other small stuff

## Autogenerated Changes
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/188
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/189
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/190
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/191
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/196
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/199
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/200
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/203
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/204
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/205
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/209
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/214
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/215
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/217
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/220
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/221
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/223
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/226
* Translations update from Hosted Weblate by @weblate in https://github.com/deniscerri/ytdlnis/pull/228

**Full Changelog**: https://github.com/deniscerri/ytdlnis/compare/v1.6.3...v1.6.4

<hr /><em>This discussion was created from the release <a href='https://github.com/deniscerri/ytdlnis/releases/tag/v1.6.4'>v1.6.4</a>.</em>

**Comment by matt873** [comment]
Tapping the notification of the errored download leads to the running tab.

**Reply by deniscerri** [reply]
noted

**Comment by matt873** [comment]
The "Embed Thumbnail" function is not working. Thought it might be an issue with yt-dlp but I tested the yt-dlp interface app on pc and tried to download the same video with the latest stable version of yt-dlp and it worked. So it might be an issue with the app, not yt-dlp.

**Reply by deniscerri** [reply]
What container are you using. Send a download log because i cannot reproduce it

**Comment by matt873** [comment]
Container is webm. Where can I find log file? I have enabled the option in settings.
![Screenshot_20230731-180717](https://github.com/deniscerri/ytdlnis/assets/108935493/660f0698-fd21-48c0-ab9e-ef91ca9e674f)
![Screenshot_20230731-180611](https://github.com/deniscerri/ytdlnis/assets/108935493/34094ce8-24c1-459a-9f65-1d7a6ef1c2e6)

**Reply by teddysulaimanGL** [reply]
Dude, that preference is only for Audio

**Reply by matt873** [reply]
It was working on previous versions. I have some video downloads from Ytdlnis that have thumbnails embedded in them. The same function can be tested with the yt-dlp interface app for Windows. It also does embed thumbnails without saving them

**Reply by matt873** [reply]
![Screenshot_20230731-182456](https://github.com/deniscerri/ytdlnis/assets/108935493/c4d39b11-9a71-4295-ae49-c60dd9d33faa)
![Screenshot_20230731-182445](https://github.com/deniscerri/ytdlnis/assets/108935493/d3e4173b-b93e-4982-a414-164546fcf04b)

**Comment by matt873** [comment]
Downloading:
Title: 4 Tips To Reduce Amazon FBA Fees in 2023
URL: https://youtube.com/watch?v=PSxqWXQyejU&feature=share
Type: video
Format: Format(format_id=243, container=WEBM, vcodec=vp9, acodec=, encoding=, filesize=11979149, format_note=360p, fps=30, asr=, url=https://pipedproxy-ams-2.kavin.rocks/videoplayback?expire=1690830690&ei=ArPHZPWrGYOYgAe-nbeIAg&ip=144.21.40.247&id=o-AG2eWsxdUPsCeOHRDOD6KfzBCSW__vCdtwX6usrDgCcS&itag=243&source=youtube&requiressl=yes&mh=Eq&mm=31%2C26&mn=sn-5hne6nzy%2Csn-4g5ednde&ms=au%2Conr&mv=m&mvi=3&pl=25&initcwndbps=211250&spc=Ul2SqzUY8K01OiY4mVdZyP8RsnsPhlc&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=11979149&ratebypass=yes&dur=375.699&lmt=1689709523932862&mt=1690808742&fvip=3&keepalive=yes&fexp=24007246%2C24362685%2C51000014%2C51000022&beids=24350017&c=ANDROID&txp=5319224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cratebypass%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAIzjUTIqLjvpac7j9QXwmDQFbHHjUVn8RaJ4UEUy7pXTAiA2Yg59pTSEgvujZmzQ-Pj0ZEFYfB6PyshxTaYStBF6Ew%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIhAO4bNpxJHm8EKVCWE-y0uX64Y0HyMKnu9u-n8GEclboDAiA3nOIVk0Fb3fg7-Slapqjv0vF-3fuG-II6xmxKXazd5Q%3D%3D&cpn=lNDmS08QjeLwIeeh&host=rr3---sn-5hne6nzy.googlevideo.com)

Command: -N "25" --sponsorblock-mark "all" --embed-chapters --embed-subs --sub-langs "en" -f "243+bestaudio/243/best" --write-subs --write-auto-subs --sub-format "srt/best" --convert-subtitles "srt" -P "/data/user/0/com.deniscerri.ytdl/cache/downloads/42" -o "%(title)s - %(uploader|Evolve Media Agency)s.%(ext)s" --replace-in-metadata "title" ".*." "4 Tips To Reduce Amazon FBA Fees in 2023" --replace-in-metadata "uploader" ".*." "Evolve Media Agency" --replace-in-metadata "uploader" " - Topic$" " " "https://youtube.com/watch?v=PSxqWXQyejU&feature=share" 

[youtube] PSxqWXQyejU: Downloading webpage
[youtube] PSxqWXQyejU: Downloading ios player API JSON
[youtube] PSxqWXQyejU: Downloading android player API JSON
[youtube] PSxqWXQyejU: Downloading m3u8 information
[info] PSxqWXQyejU: Downloading subtitles: en
[MetadataParser] Changed uploader to: Evolve Media Agency
[MetadataParser] Did not find ' - Topic$' in uploader
[SponsorBlock] Fetching SponsorBlock segments
[SponsorBlock] No matching segments were found in the SponsorBlock database
[info] PSxqWXQyejU: Downloading 1 format(s): 243+251
[info] Writing video subtitles to: /data/user/0/com.deniscerri.ytdl/cache/downloads/42/4 Tips To Reduce Amazon FBA Fees in 2023 - Evolve Media Agency.en.vtt
[download] Destination: /data/user/0/com.deniscerri.ytdl/cache/downloads/42/4 Tips To Reduce Amazon FBA Fees in 2023 - Evolve Media Agency.en.vtt
[download]    1.00KiB at  171.61KiB/s (00:00:00)
[download]    3.00KiB at  301.30KiB/s (00:00:00)
[download]    7.00KiB at  427.67KiB/s (00:00:00)
[download]   15.00KiB at  668.19KiB/s (00:00:00)
[download]   31.00KiB at  358.05KiB/s (00:00:00)
[download]   51.52KiB at  439.58KiB/s (00:00:00)
[download] 100% of   51.52KiB in 00:00:00 at 61.50KiB/s
[SubtitlesConvertor] Converting subtitles
[download] Destination: /data/user/0/com.deniscerri.ytdl/cache/downloads/42/4 Tips To Reduce Amazon FBA Fees in 2023 - Evolve Media Agency.f243.webm
[download]   0.0% of   11.42MiB at   64.01KiB/s ETA 03:04
[download]   0.0% of   11.42MiB at  136.74KiB/s ETA 01:25
[download]   0.1% of   11.42MiB at  235.08KiB/s ETA 00:49
[download]   0.1% of   11.42MiB at  399.39KiB/s ETA 00:29
[download]   0.3% of   11.42MiB at  419.10KiB/s ETA 00:27
[download]   0.5% of   11.42MiB at  408.12KiB/s ETA 00:28
[download]   1.1% of   11.42MiB at  549.26KiB/s ETA 00:21
[download]   2.2% of   11.42MiB at  825.74KiB/s ETA 00:13
[download]   4.4% of   11.42MiB at    1.23MiB/s ETA 00:08
[download]   8.7% of   11.42MiB at    1.94MiB/s ETA 00:05
[download]  17.5% of   11.42MiB at    2.71MiB/s ETA 00:03
[download]  35.0% of   11.42MiB at    3.30MiB/s ETA 00:02
[download]  70.0% of   11.42MiB at    3.65MiB/s ETA 00:00
[download]  87.1% of   11.42MiB at    3.74MiB/s ETA 00:00
[download]  87.1% of   11.42MiB at   43.53KiB/s ETA 00:34
[download]  87.1% of   11.42MiB at   78.31KiB/s ETA 00:19
[download]  87.2% of   11.42MiB at  145.04KiB/s ETA 00:10
[download]  87.2% of   11.42MiB at  268.51KiB/s ETA 00:05
[download]  87.4% of   11.42MiB at  219.30KiB/s ETA 00:06
[download]  87.7% of   11.42MiB at  365.06KiB/s ETA 00:03
[download]  88.2% of   11.42MiB at  264.79KiB/s ETA 00:05
[download]  89.3% of   11.42MiB at  294.35KiB/s ETA 00:04
[download]  91.5% of   11.42MiB at  358.54KiB/s ETA 00:02
[download]  95.4% of   11.42MiB at  451.85KiB/s ETA 00:01
[download] 100.0% of   11.42MiB at  538.06KiB/s ETA 00:00
[download] 100% of   11.42MiB in 00:00:06 at 1.65MiB/s   
[download] Destination: /data/user/0/com.deniscerri.ytdl/cache/downloads/42/4 Tips To Reduce Amazon FBA Fees in 2023 - Evolve Media Agency.f251.webm
[download]   0.0% of    5.42MiB at   26.36KiB/s ETA 03:31
[download]   0.1% of    5.42MiB at   66.52KiB/s ETA 01:23
[download]   0.1% of    5.42MiB at  130.18KiB/s ETA 00:42
[download]   0.3% of    5.42MiB at  237.52KiB/s ETA 00:23
[download]   0.6% of    5.42MiB at  414.71KiB/s ETA 00:13
[download]   1.1% of    5.42MiB at  449.94KiB/s ETA 00:12
[download]   2.3% of    5.42MiB at  489.81KiB/s ETA 00:11
[download]   4.6% of    5.42MiB at  570.70KiB/s ETA 00:09
[download]   9.2% of    5.42MiB at  624.56KiB/s ETA 00:08
[download]  18.4% of    5.42MiB at  663.40KiB/s ETA 00:06
[download]  31.2% of    5.42MiB at  655.64KiB/s ETA 00:05
[download]  42.8% of    5.42MiB at  673.33KiB/s ETA 00:04
[download]  55.9% of    5.42MiB at  725.86KiB/s ETA 00:03
[download]  73.4% of    5.42MiB at  771.84KiB/s ETA 00:01
[download]  90.9% of    5.42MiB at  823.35KiB/s ETA 00:00
[download] 100% of    5.42MiB in 00:00:07 at 739.66KiB/s 
[Merger] Merging formats into "/data/user/0/com.deniscerri.ytdl/cache/downloads/42/4 Tips To Reduce Amazon FBA Fees in 2023 - Evolve Media Agency.webm"
[Metadata] There isn't any metadata to add

**Reply by deniscerri** [reply]
The app embeds thumbnail only if it isn't webm

**Reply by matt873** [reply]
Just tried and it's working with mp4. I tried yesterday mp4 also but it was not embedding, so posted it here. Maybe I missed something at that time. Thank you.
